routes/users.js

```diff
@@ -1,73 +1,103 @@
 const express = require('express');
 const router = express.Router();
 const data = require('../data');
-const userData = data.users;
-const upload = require('../config/upload');
-var path = require('path');
-var fs = require('fs').promises;
+const artworks = data.artworks;
+const pictures = data.pictures;
+const validators = data.validators;
+const users = data.users;
 
+// public page
 router.get('/', async (req, res) => {
   try {
-    const userList = await userData.getAllUsers();
+    const userList = await users.getAllUsers();
     res.render('users/all', { users: userList });
   } catch (e) {
     res.status(500).send();
   }
 });
 
-router.get('/:id', async (req, res) => {
+// private page for user to see his/her own profile
+router.get('/private', async (req, res) => {
   try {
-    const user = await userData.getUserById(req.params.id);
-    res.render('users/single', { user: user });
+    const singleUser = await userData.getUserById(req.session.user._id);
+    res.render('users/single', { user: singleUser });
   } catch (e) {
     res.status(500).send();
   }
 });
 
+// private page for user to edit his/her own profile
+router.get('/edit', async (req, res) => {
+  //const user = await userData.getUserById(req.session.user._id);
+  //res.render('users/editprofile', { user: user });
+  res.render('users/editprofile');
+});
+
 router.post('/', async (req, res) => {
   const newUserData = req.body;
   // to-do basic check
   try {
-    const newUser = await userData.createUser(newUserData);
+    const newUser = await users.createUser(newUserData);
     res.redirect(`/users/${newUser._id}`);
   } catch (e) {
     res.status(500).send();
   }
 });
 
-router.put('/:id', async (req, res) => {
+// Validated user to update information
+router.put('/', async (req, res) => {
   let userInfo = req.body;
   // to-do basic checks
 
   try {
-    await userData.getUserById(req.params.id);
+    await users.getUserById(req.session.user._id);
   } catch (e) {
     res.status(404).json({ error: 'User not found' });
     return;
   }
 
   try {
-    const updatedUser = await userData.updateUser(req.params.id, userInfo);
-    res.redirect(`/users/${updatedUser._id}`);
+    const updatedUser = await users.updateUser(req.session.user._id, userInfo);
+    req.session.user = updatedUser;
+    res.redirect('/private');
   } catch (e) {
     res.sendStatus(500);
   }
 });
 
 router.delete('/:id', async (req, res) => {
   try {
-    await userData.getUserById(req.params.id);
+    await users.getUserById(req.params.id);
   } catch (e) {
     res.status(404).json({ error: 'User not found' });
     return;
   }
 
   try {
-    await userData.deleteUser(req.params.id);
+    await users.deleteUser(req.params.id);
     res.sendStatus(200);
   } catch (e) {
     res.Status(500);
   }
 });
 
+router.get('/portfolio/:id', async (req, res) => {
```

Comment:
Remember to always validate any parameter or input coming from the user. Like `req.params.id` here need to be validated.